src/optimization/Eliminator.cpp

```diff
@@ -407,3 +448,85 @@ void optimizations::eliminateRedundantMoves(const Module& module, Method& method
 		it.nextInMethod();
 	}
 }
+
+void optimizations::eliminateRedundantBitOp(const Module& module, Method& method, const Configuration& config) {
+	bool replaced;
+	do {
+		replaced = false;
+		auto it = method.walkAllInstructions();
+		while (!it.isEndOfMethod()) {
+			if (!it->hasSideEffects() && !it->hasPackMode() && !it->hasUnpackMode()) {
+				auto op = it.get<intermediate::Operation>();
+				if (op && op->op == OP_AND && ! op->hasUnpackMode() && ! op->hasPackMode()) {
+					op->getOutput().value().getSingleWriter();
+					// and v1, v2, v3 => and v1, v2, v4
+					// and v4, v1, v2    mov v4, v1
+					//
+					// and v1, v2, v3 => and v1, v2, v3
+					// or  v4, v1, v2    mov v4, v2
+					auto foundAnd = [&](Local *out, Local * in, InstructionWalker walker) {
+						auto it = walker.copy().nextInMethod();
```

Comment:
If you are iterating only to the end of the basic block, you should use `nextInBlock()` otherwise it could happen, that you jump into the next block here (if `walker` was the last instruction in a basic block).